packet/modify.go

```diff
@@ -0,0 +1,328 @@
+/*
+MIT License
+
+Copyright 2016 Comcast Cable Communications Management, LLC
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+*/
+
+package packet
+
+import (
+	"github.com/Comcast/gots"
+)
+
+// flags that are reserved and should not be used.
+const (
+	invalidTransportScramblingControlFlag TransportScramblingControlOptions = 1 // 01
+	invalidAdaptationFieldControlFlag     AdaptationFieldControlOptions     = 0 // 00
+)
+
+// NewPacket creates a new packet with a Null ID, sync byte, and with the adaptation field control set to payload only.
+// This function is error free.
+func NewPacket() (pkt Packet) {
+	//Default packet is the Null packet
+	pkt[0] = 0x47 // sets the sync byte
+	pkt[1] = 0x1F // equivalent to pkt.SetPID(NullPacketPid)
+	pkt[2] = 0xFF // equivalent to pkt.SetPID(NullPacketPid)
+	pkt[3] = 0x10 // equivalent to pkt.SetAdaptationFieldControl(PayloadFlag)
+	return
+}
+
+// FromBytes creates a ts packet from a slice of bytes 188 in length.
+// If the bytes provided have errors or the slice is not 188 in length,
+// then an error vill be returned along with a nill slice.
+func FromBytes(bytes []byte) (pkt Packet, err error) {
```

Comment:
This is a bit vague. Perhaps `PacketFromBytes` or `NewPacketFromBytes`

Reply:
This will read as `packet.FromBytes(b`) in user code. This is good API.

Reply:
That's a good point. The above comment can be ignored.